algorithms/kernel/svm/svm_train_thunder_impl.i

```diff
@@ -0,0 +1,339 @@
+
+namespace daal
+{
+namespace algorithms
+{
+namespace svm
+{
+namespace training
+{
+namespace internal
+{
+template <typename algorithmFPType, typename ParameterType, CpuType cpu>
+services::Status SVMTrainImpl<thunder, algorithmFPType, ParameterType, cpu>::compute(const NumericTablePtr & xTable, NumericTable & yTable,
+                                                                                     daal::algorithms::Model * r, const ParameterType * svmPar)
+{
+    DAAL_ITTNOTIFY_SCOPED_TASK(COMPUTE);
+
+    services::Status status;
+
+    const algorithmFPType C(svmPar->C);
+    const algorithmFPType eps(svmPar->accuracyThreshold);
+    const algorithmFPType tau(svmPar->tau);
+    const size_t maxIterations(svmPar->maxIterations);
+    const size_t cacheSize(svmPar->cacheSize);
+    kernel_function::KernelIfacePtr kernel = svmPar->kernel->clone();
+
+    const size_t nVectors  = xTable->getNumberOfRows();
+    const size_t nFeatures = xTable->getNumberOfColumns();
+
+    TArray<algorithmFPType, cpu> alphaTArray(nVectors);
+    DAAL_CHECK_MALLOC(alphaTArray.get());
+    algorithmFPType * alpha = alphaTArray.get();
+
+    ReadColumns<algorithmFPType, cpu> mtY(yTable, 0, 0, nVectors);
+    DAAL_CHECK_BLOCK_STATUS(mtY);
+    const algorithmFPType * y = mtY.get();
+
+    // gradi = -yi; ai = 0
+    TArray<algorithmFPType, cpu> gradTArray(nVectors);
+    DAAL_CHECK_MALLOC(gradTArray.get());
+    algorithmFPType * grad = gradTArray.get();
+
+    for (size_t i = 0; i < nVectors; i++)
+    {
+        grad[i]  = -y[i];
+        alpha[i] = algorithmFPType(0);
+    }
+
+    TaskWorkingSet<algorithmFPType, cpu> workSet(nVectors, maxBlockSize);
+    DAAL_CHECK_STATUS(status, workSet.init());
+    const size_t nWS = workSet.getSize();
+    const size_t innerMaxIterations(nWS * cInnerIterations);
+
+    algorithmFPType diff     = algorithmFPType(0);
+    algorithmFPType diffPrev = algorithmFPType(0);
+
+    size_t innerIteration = 0;
+    size_t sameLocalDiff  = 0;
+
+    TArray<algorithmFPType, cpu> buffer(nWS * MemSmoId::latest + nWS * nWS);
+    DAAL_CHECK_MALLOC(buffer.get());
+
+    TArray<algorithmFPType, cpu> deltaAlpha(nWS);
+    DAAL_CHECK_MALLOC(deltaAlpha.get());
+
+    SVMCachePtr<thunder, algorithmFPType, cpu> cachePtr;
+
+    TArray<char, cpu> I(nWS);
+    DAAL_CHECK_MALLOC(I.get());
+
+    DAAL_OVERFLOW_CHECK_BY_MULTIPLICATION(size_t, nVectors * sizeof(algorithmFPType), nVectors);
+    if (cacheSize >= nVectors * nVectors * sizeof(algorithmFPType))
+    {
+        const size_t defaultCacheSize = nWS;
+        // TODO: support the simple cache for thunder method
+        cachePtr = SVMCache<thunder, noCache, algorithmFPType, cpu>::create(defaultCacheSize, nVectors, xTable, kernel, status);
+    }
+    else
+    {
+        const size_t defaultCacheSize = nWS;
+        cachePtr                      = SVMCache<thunder, noCache, algorithmFPType, cpu>::create(defaultCacheSize, nVectors, xTable, kernel, status);
+    }
+
+    size_t iter = 0;
```

Comment:
Makes sense to combine with next line ?

Reply:
I would like to leave it because of debugging the number of iterations. In the future, I would like to return as a result.